README.md

```diff
@@ -114,101 +178,262 @@ another desired outcome. If a policy already exists for your host, you can skip
 
-```sh
-export CONJUR_ACCOUNT=<account specified during Conjur setup>
-export CONJUR_APPLIANCE_URL=<Conjur HTTPS endpoint>
-```
+As mentioned in the [Configuration](#configuration) section, you can provide varying ways
+for your app to authenticate against a Conjur server. Generally environment variables are
+most common but this isn't the only way. In addition to explicitly setting these environment
+variables, you can do so by providing [properties](#system-properties), using the Credentials
+object, or by providing an Authorization Token. Once you have chosen from one of the
+patterns below that works for you, you can now create a `Conjur` class instance values to
+access Conjur services and make RESTful API calls.
 
-but this isn't the only way. In addition to explicitly setting these environment variables, you can do so by providing values, using the Credentials object, or providing an Authorization Token)
-Once you have chosen from one of the patterns below that works for you, you can now create a `Conjur` instance from these values to access Conjur services and make RESTful API calls.
+### Environment Variables
 
-### Environment Variables (the standard way)
-```sh
-# Additionally set the following environment variables:
+```bash
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur endpoint URL>
 export CONJUR_AUTHN_LOGIN=<user/host identity>
 export CONJUR_AUTHN_API_KEY=<user/host API key>
 ```
 ```java
+import net.conjur.api.Conjur;
+
+// Using environment variables
+Conjur conjur = new Conjur();
+```
+
+### System Properties
+
+```sh-session
+$ java -jar myConjurClient.jar \
+     -DCONJUR_ACCOUNT=<account specified during Conjur setup> \
+     -DCONJUR_APPLIANCE_URL=<Conjur endpoint URL> \
+     -DCONJUR_AUTHN_LOGIN=<user/host identity> \
+     -DCONJUR_AUTHN_API_KEY=<user/host API key>
+```
+```java
+import net.conjur.api.Conjur;
+
+// Using environment variables
+Conjur conjur = new Conjur();
+```
+
+### System Properties with Maven
+
+```sh-session
+$ mvn exec:java \
+  -DCONJUR_ACCOUNT=<account specified during Conjur setup> \
+  -DCONJUR_APPLIANCE_URL=<Conjur endpoint URL> \
+  -DCONJUR_AUTHN_LOGIN=<user/host identity> \
+  -DCONJUR_AUTHN_API_KEY=<user/host API key> \
+  -Dexec.mainClass="com.myorg.client.App"
+```
+```java
+import net.conjur.api.Conjur;
+
 // Using environment variables
 Conjur conjur = new Conjur();
 ```
 
 ### Username and Password
+
+```bash
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur endpoint URL>
+```
+
 ```java
+import net.conjur.api.Conjur;
+
 // Authenticate using provided username/hostname and password/API key
 Conjur conjur = new Conjur('host/host-id', 'api-key');
+// or
 Conjur conjur = new Conjur('username', 'password');
```

Comment:
does it really work with password or API key? if it's a password, the client would have to know to login first and then authenticate with the API key, right?

maybe that is how it works, but I'm not sure from the method signature that the function would know whether it has a password or API key to make the decision about what to do, and afaik the authenticate route requires an API key

Reply:
Hmm I'll look into this - there's a confusion I think in some of our "api" vs "password" wording

Reply:
I don't have a good setup to get a host working but I think the phrasing here is:
`Conjur('host/host-id', 'api-key');` will try logging in with `host` as `user` and `api-key` as password
`Conjur conjur = new Conjur('username', 'password');` will try logging in with `username` as `user` and `password` as `password`.

Reply:
have you tried it with username / password? if it works, then I suppose this can stay as-is

Reply:
I've used the admin account (which I suppose has an "api key" even though it's realistically the password(?) ). let me make a real user and set a password to see what's going on here - I might just be confused.